inst/openvpn/openvpn.go

```diff
@@ -0,0 +1,188 @@
+package openvpn
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+)
+
+// OpenVPN has a openvpn configuration.
+type OpenVPN struct {
+	Path      string
+	Role      string
+	Tap       *tapInterface
+	Port      int
+	Proto     string
+	Host      *host
+	Managment *host
+	Server    *host
+	Service   string
+	Validity  *validity
+}
+
+type validity struct {
+	Year  int
+	Month int
+	Day   int
+}
+
+type host struct {
+	IP       string
+	Port     int
+	Mask     string
+	Protocol string
+}
+
+// NewOpenVPN creates a default OpenVPN configuration.
+func NewOpenVPN() *OpenVPN {
+	return &OpenVPN{
+		Path:  ".",
+		Tap:   &tapInterface{},
+		Role:  "server",
+		Proto: "udp",
+		Host: &host{
+			IP:       "0.0.0.0",
+			Port:     443,
+			Protocol: "tcp",
+		},
+		Managment: &host{
+			IP:       "127.0.0.1",
+			Port:     7505,
+			Protocol: "tcp",
+		},
+		Server: &host{
+			IP:   "10.217.3.0",
+			Mask: "255.255.255.0",
+		},
+		Validity: &validity{
+			Year: 10,
+		},
+	}
+}
+
+// InstallTap installs a new tap interface.
+func (o *OpenVPN) InstallTap() (err error) {
+	o.Tap, err = installTAP(o.Path, o.Role)
+	return err
+}
+
+// RemoveTap removes the tap interface.
+func (o *OpenVPN) RemoveTap() (err error) {
+	return o.Tap.remove(o.Path)
+}
+
+// Configurate configurates openvpn config files.
+func (o *OpenVPN) Configurate() error {
+	if o.isClient() {
+		return nil
+	}
+
+	if err := o.createSertificate(); err != nil {
+		return err
+	}
+
+	file, err := os.Create(filepath.Join(o.Path, "config/"+o.Role+".conf"))
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	templ, err := template.New("ovpnTemplate").Parse(serverTemplate)
+	if err != nil {
+		return err
+	}
+
+	// Set dynamic port.
+	o.Managment.Port = freePort(*o.Managment)
+	o.Host.Port = freePort(*o.Host)
+
+	return templ.Execute(file, &o)
+}
+
+// RemoveConfig removes openvpn configuration.
+func (o *OpenVPN) RemoveConfig() error {
+	if o.isClient() {
+		return nil
+	}
+
+	os.Remove(filepath.Join(o.Path, "config/dh2048.pem"))
+	os.Remove(filepath.Join(o.Path, "config/ca.crt"))
+	os.Remove(filepath.Join(o.Path, "config/ca.key"))
+	os.Remove(filepath.Join(o.Path, "config/"+o.Role+".crt"))
+	os.Remove(filepath.Join(o.Path, "config/"+o.Role+".key"))
+	os.Remove(filepath.Join(o.Path, "config/"+o.Role+".conf"))
```

Comment:
Well, I would have done it in another way. 

(sorry for the syntax)

```suggestion
pathsToRemove = [
    "config/dh2048.pem",
    "config/ca.crt",
    "config/ca.key",
    "config/"+o.Role+".crt",
] 

for path in pathsToRemove{
    os.Remove(filepath.Join(o.Path, path))
}

```